### PR TITLE
test: Add ramps URL scheme deeplinking e2e

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

The purpose of this PR is to add e2e coverage for the URL schema deeplinks in the ramps flow. 

The goal here is to ensure that we properly test the URL scheme deep linking in these flows. 

### Sell Deep link

```
Scenario 1
Given I open a sell ETH deeplink for mainnet 
Then the app should launch with the ramps build quotes page
And the token I want to sell is displayed correctly
And the amount i want to send is displayed correctly

Scenario 2
Given I open a sell deeplink on an unsupported network
Then the app should launch with the ramps build quotes page
And I am prompted to add the network
When I add the network
Then the Quotes page should be displayed 
```

### Buy deeplink Flow

```
Scenario 3
Given i deeplink to the buy eth flow on mainnet
Then the app should launch with the ramps build quotes page
And the token I want to sell is displayed correctly
And the amount i want to send is displayed correctly

Scenario 4
Given i deeplink to the buy eth flow on a popular network
Then the app should launch with the ramps build quotes page
And the token I want to sell is displayed correctly
And the amount i want to send is displayed correctly

```
## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: e2e/fixtures/fixture-builder.js [hunk 1]
```diff
@@ -776,6 +779,24 @@ class FixtureBuilder {
     return this;
   }
 
+  withRampsSelectedRegion() {
```
Comment (cortisiko) [inline e2e/fixtures/fixture-builder.js:782]: Created a state where my ramps region is set. This allows me to skip through the ramps onboarding flow

File: e2e/fixtures/fixture-builder.js [hunk 2]
```diff
@@ -856,9 +878,10 @@ class FixtureBuilder {
     const sepoliaConfig = CustomNetworks.Sepolia.providerConfig;
 
     // Generate a unique key for the new network client ID
-    const newNetworkClientId = `networkClientId${Object.keys(fixtures.NetworkController.networkConfigurationsByChainId)
-      .length + 1
-      }`;
+    const newNetworkClientId = `networkClientId${
```
Comment (cortisiko) [inline e2e/fixtures/fixture-builder.js:881]: My editor formatted some of these. Nothing new was added

File: e2e/fixtures/fixture-builder.js [hunk 3]
```diff
@@ -908,8 +931,9 @@ class FixtureBuilder {
       } = network.providerConfig;
 
       // Generate a unique key for the new network client ID
-      const newNetworkClientId = `networkClientId${Object.keys(networkConfigurationsByChainId).length + 1
-        }`;
+      const newNetworkClientId = `networkClientId${
```
Comment (cortisiko) [inline e2e/fixtures/fixture-builder.js:934]: My editor formatted some of these. Nothing new was added

File: e2e/fixtures/fixture-builder.js [hunk 4]
```diff
@@ -988,19 +1012,16 @@ class FixtureBuilder {
...
   }
 
   withIncomingTransactionPreferences(incomingTransactionPreferences) {
-    merge(
-      this.fixture.state.engine.backgroundState.PreferencesController,
-      {
-        showIncomingTransactions: incomingTransactionPreferences,
-      },
-    );
+    merge(this.fixture.state.engine.backgroundState.PreferencesController, {
```
Comment (cortisiko) [inline e2e/fixtures/fixture-builder.js:1022]: My editor formatted some of these. Nothing new was added

File: e2e/pages/Ramps/BuildQuoteView.js [hunk 5]
```diff
@@ -22,6 +22,10 @@ class BuildQuoteView {
   async tapCancelButton() {
     await Gestures.waitAndTap(this.cancelButton);
   }
+  async tapDefaultToken(token) {
```
Comment (cortisiko) [inline e2e/pages/Ramps/BuildQuoteView.js:25]: Tapping on the default token in the build quotes screen

File: e2e/pages/Ramps/TokenSelectBottomSheet.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import Matchers from '../../utils/Matchers';
```
Comment (cortisiko) [inline e2e/pages/Ramps/TokenSelectBottomSheet.js:1]: Creating a page object for the token select bottom sheet

File: e2e/specs/confirmations/signatures/security-alert-signatures.mock.spec.js [hunk 7]
```diff
@@ -12,7 +12,7 @@ import { mockEvents } from '../../../api-mocking/mock-config/mock-events';
 import ConfirmationView from '../../../pages/Confirmation/ConfirmationView';
 import { SmokeConfirmations } from '../../../tags';
 
-describe(SmokeConfirmations('Security Alert API - Signature'), () => {
+describe('Security Alert API - Signature', () => {
```
Comment (cortisiko) [inline e2e/specs/confirmations/signatures/security-alert-signatures.mock.spec.js:15]: This is a temp change. Will revert prior to merging my PR

File: e2e/specs/confirmations/security-alert-send-eth.mock.spec.js [hunk 8]
```diff
@@ -13,7 +13,7 @@ import { mockEvents } from '../../api-mocking/mock-config/mock-events';
 import Assertions from '../../utils/Assertions';
 import { SmokeConfirmations } from '../../tags';
 
-describe(SmokeConfirmations('Security Alert API - Send flow'), () => {
+describe('Security Alert API - Send flow', () => {
```
Comment (cortisiko) [inline e2e/specs/confirmations/security-alert-send-eth.mock.spec.js:16]: This is a temp change. Will revert prior to merging my PR

File: e2e/fixtures/fixture-builder.js [hunk 9]
```diff
@@ -447,7 +448,7 @@ class FixtureBuilder {
           whitelist: [],
           tabs: [
             {
-              url: 'https://portfolio.metamask.io/explore?MetaMaskEntry=mobile/',
+              url: 'https://metamask.github.io/test-dapp/',
```
Comment (wachunei) [inline e2e/fixtures/fixture-builder.js:451]: does this affect all test? does it have an impact?

File: e2e/fixtures/fixture-builder.js [hunk 10]
```diff
@@ -776,6 +779,24 @@ class FixtureBuilder {
...
+    const defaultRegion = {
+      currencies: ['/currencies/fiat/xcd'],
+      emoji: '🇱🇨',
+      id: '/regions/lc',
+      name: 'Saint Lucia',
+      support: { buy: true, sell: true, recurringBuy: true },
+      unsupported: false,
+      recommended: false,
+      detected: false,
+      selectedPaymentMethodAgg: '/payments/debit-credit-card',
```
Comment (wachunei) [inline e2e/fixtures/fixture-builder.js:792]: This is not part of the region interface, it belongs to the `fiatOrders` state (`state.fiatOrders.selectedPaymentMethodAgg`)

File: e2e/specs/ramps/deeplink-to-sell-flow.spec.js [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+    const franceRegion = {
+      currencies: ['/currencies/fiat/eur'],
+      emoji: '🇫🇷',
+      id: '/regions/fr',
+      name: 'France',
+      support: { buy: true, sell: true, recurringBuy: true },
+      unsupported: false,
+      recommended: false,
+      detected: false,
+      selectedPaymentMethodAgg: '/payments/debit-credit-card',
```
Comment (wachunei) [inline e2e/specs/ramps/deeplink-to-sell-flow.spec.js:37]: same as above, `selectedPaymentMethodAgg` is outside the region and is not read in the code form here. Probably we want a method `withRampsSelectedPaymentMethod(paymentMethodId: string)` if we want to override the default one.

File: e2e/fixtures/fixture-builder.js (hunk 10 above)
Comment (cortisiko) [reply e2e/fixtures/fixture-builder.js:792]: Ahh good call out. Thank you. Will update

File: e2e/fixtures/fixture-builder.js (hunk 9 above)
Comment (cortisiko) [reply e2e/fixtures/fixture-builder.js:451]: ahh this is definitely not intended on this PR. The idea is to let all the e2e tests use the test dapp as the default site. i will remove this change

File: e2e/specs/ramps/deeplink-to-sell-flow.spec.js (hunk 11 above)
Comment (cortisiko) [reply e2e/specs/ramps/deeplink-to-sell-flow.spec.js:37]: Good point, although it is not need for these e2e,  I will create a method just to update the payment type

**Review by wachunei — APPROVED** [review]

**Pull request merged**